src/app/http-stalker.service.ts

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Injectable} from '@angular/core';
+import {User} from './user';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class HttpStalkerService {
+  httpOptions: any = {
+    headers: new HttpHeaders({'Content-Type': 'application/json'}),
+    observe: 'response',
+  };
```

Comment:
qui l'any proprio non ci va. dato che è un object literal, il tipo è semplicemente
```ts
interface HttpOptions {
  headers: HttpHeaders;
  observe: string;
}
```

ma viene inferito automaticamente da typescript.
invece il tipo di ritorno di this.http.post è ` Observable<HttpResponse<TIPO DI OGGETTO RESTITUITO>>`

Reply:
Questo se il compilatore prendesse la versione giusta di post, ma siccome il tipo di ritorno dipende dalle httpOptions che gli passo la versione che il compilatore vede è quella di default con Observable<arraybuffer>
Se cerco di specificare il tipo in quella funziona non compila

Reply:
se vuoi domani ci mettiamo insieme a risolverlo in questo caso specifico!